File: services/dashboard-ui/client/lib/ctl-api/installs/components/deploy-components.ts
import { api } from '@/lib/api'
import type { TWorkflowResponse } from '@/types'

export type TDeployComponentsBody = {
  plan_only?: boolean
  role?: string
}

export const deployComponents = ({
  installId,
  orgId,
  body,
}: {
  installId: string
  orgId: string
  body: TDeployComponentsBody
}) =>
  api<TWorkflowResponse>({
    withHeaders: true,
    path: `installs/${installId}/components/deploy-all`,
    method: 'POST',
    orgId,
    body,
  })

File: bins/cli/internal/services/installs/deploy_components.go
package installs

import (
	"context"
	"fmt"

	"github.com/nuonco/nuon/bins/cli/internal/lookup"
	"github.com/nuonco/nuon/bins/cli/internal/ui"
)

func (s *Service) DeployComponents(ctx context.Context, installID string, roleName string, planOnly bool, asJSON bool) error {
	installID, err := lookup.InstallID(ctx, s.api, installID)
	if err != nil {
		return ui.PrintError(err)
	}

	_, err = s.api.DeployInstallComponents(ctx, installID, roleName, planOnly)
	if err != nil {
		fmt.Printf("deploy components err: %+s\n", err)
		return ui.PrintJSONError(err)
	}

	ui.PrintLn("successfully triggered deploy of all install components")
	return nil
}
